source/jormungandr/jormungandr/realtime_schedule/timeo.py

```diff
@@ -196,6 +199,17 @@ def _get_passages(self, timeo_resp, current_dt, line_uri=None):
             'timeo response: {}'.format(timeo_resp), extra={'rt_system_id': unicode(self.rt_system_id)}
         )
 
+        # internal timeo error handling
+        message_responses = timeo_resp.get('MessageResponse')
+        for message_response in message_responses:
+            if message_response['ResponseCode'] >= self.INTERNAL_TIMEO_ERROR_CODE_LIMIT:
```

Comment:
for safety
```suggestion
            if 'ResponseCode' in message_response and message_response['ResponseCode'] >= self.INTERNAL_TIMEO_ERROR_CODE_LIMIT:
```
or use a get?